### PR TITLE
Incentives UI

### PR DESCRIPTION
**Pull request by gmxer**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205829386512415

### CODE REVIEW

File: src/domain/synthetics/markets/useMarketTokensAPR.ts [hunk 1]
```diff
@@ -28,18 +33,35 @@ export function useMarketTokensAPR(chainId: number): MarketTokensAPRResult {
...
       const nowInSecods = Math.floor(Date.now() / 1000);
 
+      let arbTokenAddress: null | string = null;
+      try {
+        arbTokenAddress = getTokenBySymbol(chainId, "ARB").address;
+      } catch (err) {}
+      let arbTokenPrice = BigNumber.from(0);
+
+      if (arbTokenAddress) {
+        arbTokenPrice = tokensData
```
Comment (gdev8317) [inline src/domain/synthetics/markets/useMarketTokensAPR.ts:58]: maybe remove ternary operator?:
```
if (arbTokenAddress && tokensData) {
  arbTokenPrice = tokensData[arbTokenAddress]?.prices?.minPrice || BigNumber.from(0)
}
```

looks a bit simpler

File: src/domain/synthetics/markets/useMarketTokensAPR.ts [hunk 2]
```diff
@@ -99,20 +126,46 @@ export function useMarketTokensAPR(chainId: number): MarketTokensAPRResult {
         return acc;
       }, {} as MarketTokensAPRData);
 
-      const avgMarketsAPR = Object.values(marketTokensAPRData)
+      const marketsTokensIncentiveAprData: MarketTokensAPRData = !shouldCalcBonusApr
+        ? {}
+        : marketAddresses.reduce((acc, marketAddress) => {
+            if (!rawIncentivesStats) return acc;
```
Comment (gdev8317) [inline src/domain/synthetics/markets/useMarketTokensAPR.ts:132]: is it better to move this check outside?

```
const marketsTokensIncentiveAprData: MarketTokensAPRData = !shouldCalcBonusApr || ! rawIncentivesStats ? ... : ...

```

File: src/domain/synthetics/markets/useMarketTokensAPR.ts [hunk 3]
```diff
@@ -99,20 +126,46 @@ export function useMarketTokensAPR(chainId: number): MarketTokensAPRResult {
...
+
+            const arbTokensAmount = BigNumber.from(rawIncentivesStats.rewardsPerMarket[marketAddress] ?? 0);
+            const yearMultiplier = Math.floor((365 * 24 * 60 * 60) / rawIncentivesStats.period);
+            const poolValue = getByKey(marketsInfoData, marketAddress)?.poolValueMin;
+            let incentivesApr = BigNumber.from(0);
+
+            if (poolValue && poolValue.gt(0)) {
```
Comment (gdev8317) [inline src/domain/synthetics/markets/useMarketTokensAPR.ts:139]: could be just `poolValue?.gt(0)`

File: src/domain/synthetics/markets/useMarketTokensAPR.ts [hunk 4]
```diff
@@ -99,20 +126,46 @@ export function useMarketTokensAPR(chainId: number): MarketTokensAPRResult {
...
+              incentivesApr = arbTokensAmount
+                .mul(arbTokenPrice)
+                .div(poolValue)
```
Comment (gdev8317) [inline src/domain/synthetics/markets/useMarketTokensAPR.ts:142]: wondering if it works correctly

`arbTokensAmount` – 18 decimals
`arbTokenPrice` – 12 decimals
`poolValue` – 30 decimals

at first glance it should return 0

File: src/domain/synthetics/tokens/useOracleKeeperFetcher.ts [hunk 5]
```diff
@@ -127,11 +136,22 @@ export function useOracleKeeperFetcher(chainId: number) {
...
     }
 
+    async function fetchIncentivesRewards(): Promise<RawIncentivesStats> {
+      return fetch(
+        buildUrl(oracleKeeperUrl!, "/incentives/stip/lp", {
+          ignoreStartDate: forceIncentivesActive ? "1" : undefined,
+        })
+      )
+        .then((res) => res.json())
+        .catch(() => null);
```
Comment (gdev8317) [inline src/domain/synthetics/tokens/useOracleKeeperFetcher.ts:146]: does it make sense to log error? we do this for other requests

File: src/domain/synthetics/markets/useMarketTokensAPR.ts (hunk 4 above)
Comment (gmxer) [reply src/domain/synthetics/markets/useMarketTokensAPR.ts:142]: arbTokenPrice is 30 decimals, we somewhere convert it

File: src/domain/synthetics/tokens/useOracleKeeperFetcher.ts (hunk 5 above)
Comment (gmxer) [reply src/domain/synthetics/tokens/useOracleKeeperFetcher.ts:146]: we for some reason entirely ignore subgraph error here which I postpone to change. Probably needs to be updated here as well

Comment (gdev8317) [reply src/domain/synthetics/tokens/useOracleKeeperFetcher.ts:146]: not sure i got it, this is a request to our API, not subgraph

Comment (gmxer) [reply src/domain/synthetics/tokens/useOracleKeeperFetcher.ts:146]: I mean we have two calls:
```
[
   client!.query({ query: gql(`{${queryBody}}`), fetchPolicy: "no-cache" }),
   shouldCalcBonusApr ? oracleKeeperFetcher.fetchIncentivesRewards() : Promise.resolve(null),
]
```
Both of them now fails silently

**Review by gdev8317 — DISMISSED** [review]

**Review by vipineth — APPROVED** [review]

**Pull request merged**